File: java/eyes.ufg.java/src/main/java/com/applitools/eyes/visualgrid/model/EmulationBaseInfo.java
package com.applitools.eyes.visualgrid.model;

import com.applitools.eyes.RectangleSize;

public abstract class EmulationBaseInfo {

    ScreenOrientation screenOrientation;
    RectangleSize size = null;

    EmulationBaseInfo(ScreenOrientation screenOrientation) {
        this.screenOrientation = screenOrientation;
    }

    public ScreenOrientation getScreenOrientation() {
        return screenOrientation;
    }

    public void setScreenOrientation(ScreenOrientation screenOrientation) {
        this.screenOrientation = screenOrientation;
    }

    public abstract String getDeviceName();

    public RectangleSize getSize() {
        return size;
    }
}

File: dotnet/Eyes.Images.DotNet/VisualGrid/EmulationBaseInfo.cs
using Applitools.Utils.Geometry;
using System;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace Applitools.VisualGrid
{
    public abstract class EmulationBaseInfo : IEquatable<EmulationBaseInfo>
    {
        public EmulationBaseInfo(ScreenOrientation screenOrientation)
        {
            ScreenOrientation = screenOrientation;
        }

        [JsonConverter(typeof(StringEnumConverter))]
        public ScreenOrientation ScreenOrientation { get; set; }
        public RectangleSize Size { get; set; }

        public bool Equals(EmulationBaseInfo other)
        {
            if (other == null) return false;
            return GetType() == other.GetType() && 
                   ScreenOrientation == other.ScreenOrientation;
        }
    }
}